src/System.Text.Json/tests/Serialization/SnakeCaseUnitTests.cs

```diff
@@ -0,0 +1,141 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+using Xunit;
+
+namespace System.Text.Json.Serialization.Tests
+{
+    public static class SnakeCaseUnitTests
+    {
+        [Fact]
+        public static void Convert_SpecifiedName_MatchesExpectedName()
+        {
+            Assert.Equal("url_value", ConvertToSnakeCase("URLValue"));
```

Comment:
test case for a string that has a leading underscore?

Reply:
(and multiple leading underscores)